File: manager/internal/model/agent.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// Agent Agent模型
type Agent struct {
	ID        uint           `gorm:"primarykey" json:"id"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	// NodeID 节点ID(外键关联到nodes表)
	NodeID string `gorm:"index;size:50;not null" json:"node_id"`

	// AgentID Agent唯一标识符(在节点内唯一)
	AgentID string `gorm:"index;size:100;not null" json:"agent_id"`

	// Type Agent类型(filebeat/telegraf/node_exporter等)
	Type string `gorm:"size:50" json:"type"`

	// Status 运行状态(running/stopped/error/starting/stopping)
	Status string `gorm:"size:20;not null;default:'stopped'" json:"status"`

	// PID 进程ID(0表示未运行)
	PID int `gorm:"default:0" json:"pid"`

	// LastHeartbeat 最后心跳时间
	LastHeartbeat *time.Time `json:"last_heartbeat"`

	// LastSyncTime 最后同步时间
	LastSyncTime time.Time `gorm:"not null" json:"last_sync_time"`
}

// TableName 指定表名
func (Agent) TableName() string {
	return "agents"
}

// IsRunning Agent是否正在运行
func (a *Agent) IsRunning() bool {
	return a.Status == "running"
}

File: web/src/types/agent.ts
/**
 * Agent 相关类型定义
 */

export interface Agent {
  id: number;
  node_id: string;
  agent_id: string;
  type: string; // 如 "filebeat", "telegraf", "node_exporter"
  version?: string;
  status: string; // "running", "stopped", "error", "starting", "stopping"
  config?: string; // JSON 格式的配置
  pid?: number;
  last_heartbeat?: string; // ISO 8601 格式时间戳
  last_sync_time?: string; // ISO 8601 格式时间戳
  created_at: string; // ISO 8601 格式时间戳
  updated_at: string; // ISO 8601 格式时间戳
}

export type AgentOperation = 'start' | 'stop' | 'restart';

export interface OperateAgentRequest {
  operation: AgentOperation;
}

export interface AgentLogsResponse {
  logs: string[];
  count: number;
}

export interface AgentListResponse {
  agents: Agent[];
  count: number;
}

/**
 * Agent 资源数据点（与后端 ResourceDataPoint 对应）
 */
export interface AgentResourceDataPoint {
  timestamp: number; // Unix 时间戳（秒）
  cpu: number; // CPU 使用率（百分比）
  memory_rss: number; // 内存占用 RSS（字节）
  memory_vms: number; // 内存占用 VMS（字节）
  disk_read_bytes: number; // 磁盘读取字节数
  disk_write_bytes: number; // 磁盘写入字节数
  open_files: number; // 打开文件数
}

/**
 * Agent 历史指标响应
 */
export interface AgentMetricsHistoryResponse {
  agent_id: string;
  data_points: AgentResourceDataPoint[];
  count: number;
}
